### PR TITLE
Fix a problem and add some test cases for podman run security-opt flag

### PR DESCRIPTION
**Pull request by ypu**
Fix a problem for selinux check command and add 5 test cases for security-opt flag.

### CODE REVIEW

File: test/e2e/run_test.go [hunk 1]
```diff
@@ -61,6 +63,77 @@ var _ = Describe("Podman run", func() {
...
+	It("podman run seccomp test", func() {
+		selinux := podmanTest.SystemExec("ls", []string{"/usr/sbin/selinuxenabled"})
+		selinux.WaitWithDefaultTimeout()
+		if selinux.ExitCode() != 0 {
+			Skip("SELinux not enabled")
+		}
+
+		jsonFile := filepath.Join(podmanTest.TempDir, "seccomp.json")
+		in := []byte(`{"defaultAction":"SCMP_ACT_ALLOW","syscalls":[{"name":"getcwd","action":"SCMP_ACT_ERRNO"}]}`)
+		WriteJsonFile(in, jsonFile)
```
Comment (TomSweeneyRedHat) [inline test/e2e/run_test.go]: Should you error out here if WriteJsonFile fails?

File: test/e2e/libpod_suite_test.go [hunk 2]
```diff
@@ -629,6 +629,18 @@ func IsCommandAvailable(command string) bool {
...
 }
 
+// WriteJsonFile write json format data to a json file
+func WriteJsonFile(data []byte, filePath string) bool {
+	var jsonData map[string]interface{}
+	json.Unmarshal(data, &jsonData)
+	formatJson, _ := json.MarshalIndent(jsonData, "", "	")
+	err := ioutil.WriteFile(filePath, formatJson, 0644)
+	if err != nil {
+		return false
```
Comment (TomSweeneyRedHat) [inline test/e2e/libpod_suite_test.go]: It would probably be good to have at least an informational message here on error.

**Comment by rhatdan** [comment]
I like it, fix the two issues that @TomSweeneyRedHat Points out and we can add this.

File: test/e2e/libpod_suite_test.go (hunk 2 above)
Comment (ypu) [reply test/e2e/libpod_suite_test.go]: Thanks for comments. Will return the err directly in this function. And let the test cases to decide to how to deal with it.

File: test/e2e/run_test.go (hunk 1 above)
Comment (ypu) [reply test/e2e/run_test.go]: Yes, Thanks a lot for reminding. Will skip the tests if write seccomp.json failed.

**Comment by ypu** [comment]
Hi @rhatdan 
Update the test case based on the comments from @TomSweeneyRedHat . But seems it failed in the CI env. It works in my test machine which is f28 and rhel7. Will try to find similar env as CI and try them again.

File: test/e2e/run_test.go [hunk 3]
```diff
@@ -51,6 +51,7 @@ var _ = Describe("Podman run", func() {
 
 	It("podman run selinux grep test", func() {
 		selinux := podmanTest.SystemExec("ls", []string{"/usr/sbin/selinuxenabled"})
+		selinux.WaitWithDefaultTimeout()
```
Comment (rhatdan) [inline test/e2e/run_test.go]: We should be able to do this much easier and more sensible by just importing go-selinux and then just checking if selinux enabled.  No need to exec a program.

Comment (ypu) [reply test/e2e/run_test.go]: OK, will update this.

**Comment by mheon** [comment]
Tests seem green aside from known flakes on cloud podman.

File: test/e2e/run_test.go [hunk 4]
```diff
@@ -61,6 +62,70 @@ var _ = Describe("Podman run", func() {
...
+			Skip("SELinux not enabled")
+		}
+		session := podmanTest.Podman([]string{"run", "-it", "--security-opt", "label=type:spc_t", ALPINE, "cat", "/proc/self/attr/current"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		match, _ := session.GrepString("spc_t")
+		Expect(match).Should(BeTrue())
+	})
+
+	It("podman run seccomp undefine test", func() {
```
Comment (mheon) [inline test/e2e/run_test.go]: This and the other Seccomp test shouldn't required the `selinux.GetEnabled()` check - Seccomp should work without SELinux.

**Comment by mheon** [comment]
One small thing, otherwise LGTM

**Comment by rhatdan** [comment]
@ypu Remove the SELinux checks on the seccomp and then we will merge.

**Comment by rhatdan** [comment]
@ypu Rebase also, so tests will run on f28

File: test/e2e/run_test.go (hunk 4 above)
Comment (ypu) [reply test/e2e/run_test.go]: OK, will remove this. Thanks a lot.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**